src/caret_analyze/plot/bokeh/util.py

```diff
@@ -77,4 +93,171 @@ def get_callback_param_desc(callback: CallbackBase):
     if isinstance(callback, SubscriptionCallback):
         return f'topic_name: {callback.subscribe_topic_name}'
 
-    raise UnsupportedTypeError('callback type must be [ TimerCallback/ SubscriptionCallback]')
+    raise UnsupportedTypeError('callback type must be '
+                               '[ TimerCallback/ SubscriptionCallback]')
+
+
+PlotObjectTypes = Union[CallbackBase, Communication, Publisher, Subscription]
+
+
+def get_range(
+    plot_objects: Sequence[PlotObjectTypes]
+) -> Tuple[int, int]:
+    """
+    Get measurement duration.
+
+    Parameters
+    ----------
+    plot_objects: Sequence[Union[CallbackBase, Communication]]
+
+    Returns
+    -------
+    Tuple[int, int]
+        The timestamp of measurement start and measurement end
+
+    """
+    po_valid = [po for po in plot_objects if len(po.to_records()) > 0]
+    if len(po_valid) == 0:
+        logger.warning('Failed to found measurement results.')
+        return 0, 1
+
+    po_dfs = [po.to_dataframe() for po in plot_objects]
```

Comment:
ここ、to_dataframeがNaNを返すケースがあります。
NaNが含まれると後のmin/maxでエラーになってしまうので、remove_dropped=Trueとしてください。

```suggestion
    po_dfs = [po.to_dataframe(remove_dropped=True) for po in plot_objects]
```